commatrix/generate.go

```diff
@@ -0,0 +1,226 @@
+package commatrix
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clientutil "github.com/openshift-kni/commatrix/client"
+	"github.com/openshift-kni/commatrix/consts"
+	"github.com/openshift-kni/commatrix/debug"
+	"github.com/openshift-kni/commatrix/ss"
+	"github.com/openshift-kni/commatrix/types"
+)
+
+func GenerateMatrix(kubeconfig, customEntriesPath, customEntriesFormat, format string, env Env, deployment Deployment, destDir string) (m1, m2 *types.ComMatrix, err error) {
+	mat, err := New(kubeconfig, customEntriesPath, customEntriesFormat, env, deployment)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create the communication matrix: %s", err)
+	}
+
+	cs, err := clientutil.New(kubeconfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	tcpFile, udpFile, err := createOutputFiles(destDir)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer tcpFile.Close()
+	defer udpFile.Close()
+
+	nodesList, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ssComMat, err := getSSmatrix(cs, nodesList, tcpFile, udpFile)
+	if err != nil {
+		return nil, nil, err
+	}
+	return mat, ssComMat, err
+}
+
+// WriteMatsToFiles write all data to files(ss and endpoint slice matrix and the diff matrix).
+func WriteMatsToFiles(mat, ssComMat *types.ComMatrix, format string, env Env, deployment Deployment, destDir string) error {
+	printFn, err := getPrintFunction(format)
+	if err != nil {
+		return err
+	}
+
+	err = writeMatrixToFileByType(*mat, "communication-matrix", format, deployment, printFn, destDir)
```

Comment:
I don't think both writeMatrixToFileByType and getPrintFunction are needed, you can just print the mat where you check the type

Reply:
lets keep it as is